pkg/cmd/agent-task/capi/sessions.go

```diff
@@ -0,0 +1,208 @@
+	"fmt"
+	"net/http"
+	"slices"
+	"strconv"
+	"time"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+// session is an in-flight agent task
+type session struct {
+	ID            string    `json:"id"`
+	Name          string    `json:"name"`
+	UserID        uint64    `json:"user_id"`
+	AgentID       int64     `json:"agent_id"`
+	Logs          string    `json:"logs"`
+	State         string    `json:"state"`
+	OwnerID       uint64    `json:"owner_id"`
+	RepoID        uint64    `json:"repo_id"`
+	ResourceType  string    `json:"resource_type"`
+	ResourceID    int64     `json:"resource_id"`
+	LastUpdatedAt time.Time `json:"last_updated_at,omitempty"`
+	CreatedAt     time.Time `json:"created_at,omitempty"`
+	CompletedAt   time.Time `json:"completed_at,omitempty"`
+	EventURL      string    `json:"event_url"`
+	EventType     string    `json:"event_type"`
+}
+
+// A shim of a full pull request because looking up by node ID
+// using the full api.PullRequest type fails on unions (actors)
+type sessionPullRequest struct {
+	ID             string
+	FullDatabaseID string
+	Number         int
+	Title          string
+	State          string
+	URL            string
+	Body           string
+
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	ClosedAt  *time.Time
+	MergedAt  *time.Time
+
+	// Uncomment one of these to see error
+	// Author api.Author
+	// MergedBy   *api.Author
+	Repository *api.PRRepository
+}
+
+// Session is a hydrated in-flight agent task
+type Session struct {
+	session
+	PullRequest *api.PullRequest `json:"-"`
+}
+
+// ListSessionsForViewer lists all agent sessions for the
+// authenticated user up to limit.
+func (c *CAPIClient) ListSessionsForViewer(ctx context.Context, limit int) ([]*Session, error) {
+	url := baseCAPIURL + "/agents/sessions"
+
+	var sessions []session
+	page := 1
+	perPage := 50
+
+	for {
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+		if err != nil {
+			return nil, err
+		}
+
+		q := req.URL.Query()
+		q.Set("page_size", strconv.Itoa(perPage))
+		q.Set("page_number", strconv.Itoa(page))
+		req.URL.RawQuery = q.Encode()
+
+		res, err := c.httpClient.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("failed to list sessions: %s", res.Status)
+		}
+		var response struct {
+			Sessions []session `json:"sessions"`
+		}
+		if err := json.NewDecoder(res.Body).Decode(&response); err != nil {
+			return nil, fmt.Errorf("failed to decode sessions response: %w", err)
+		}
+		if len(response.Sessions) == 0 || len(sessions) >= limit {
+			break
+		}
+		sessions = append(sessions, response.Sessions...)
+		page++
+	}
+
+	// Drop any above the limit
+	if len(sessions) > limit {
+		sessions = sessions[:limit]
+	}
+
+	// Hydrate the Sessions with pull request data.
+	Sessions, err := c.hydrateSessionPullRequests(sessions)
+	if err != nil {
+		return nil, err
+	}
+
+	return Sessions, nil
```

Comment:
Var name is capitalised.


```suggestion
	result, err := c.hydrateSessionPullRequests(sessions)
	if err != nil {
		return nil, err
	}

	return result, nil
```

Reply:
Fixed in 13c293f4cfb08f8af46936d5d3469969005b1fa8